libsolidity/formal/SMTEncoder.cpp

```diff
@@ -970,13 +970,25 @@ void SMTEncoder::endVisit(IndexAccess const& _indexAccess)
 
 	if (_indexAccess.annotation().type->category() == Type::Category::TypeType)
 		return;
-	if (_indexAccess.baseExpression().annotation().type->category() == Type::Category::FixedBytes)
+	if (auto const* type = dynamic_cast<FixedBytesType const*>(_indexAccess.baseExpression().annotation().type))
 	{
-		m_errorReporter.warning(
-			7989_error,
-			_indexAccess.location(),
-			"Assertion checker does not yet support index accessing fixed bytes."
-		);
+		smtutil::Expression base = expr(_indexAccess.baseExpression());
+
+		if (type->numBytes() == 1)
+			defineExpr(_indexAccess, base);
+		else
+		{
+			auto [bvSize, isSigned] = smt::typeBvSizeAndSignedness(_indexAccess.baseExpression().annotation().type);
+			solAssert(!isSigned, "");
+
+			smtutil::Expression idx = expr(*_indexAccess.indexExpression());
+
+			auto bvBase = smtutil::Expression::int2bv(base, bvSize);
+			auto bvShl = smtutil::Expression::int2bv(idx * 8, bvSize);
+			auto bvShr = smtutil::Expression::int2bv(bvSize - 8, bvSize);
+			auto result = (bvBase << bvShl) >> bvShr;
```

Comment:
Will this work if it shifts outside the value range? e.g. indexing out of bounds.

Reply:
What should it answer? It mostly answers that the result is 0, but for
```
    function f(uint i) public pure {
        bytes4 x = 0x01020304;
        require(i > 3);
        assert(x[i] == 0x00);
    }
```
it says
```
Warning: Assertion violation happens here.
 --> ../test/libsolidity/smtCheckerTests/types/fixed_bytes_access_7.sol:7:3:
  |
7 | 		assert(x[i] == 0x00);
  | 		^^^^^^^^^^^^^^^^^^^^
Note: 
Counterexample:

i = 536870913


Transaction trace:
f(536870913)
```

Reply:
Maybe we can just answer 0 for now and soon add checks for bounds, because that's what's in runtime anyway.

Reply:
So I updated the code to return an `unknown` value for out of bound access, I think it's the safest option for now.

Reply:
Sounds good, the question was whether it will crash or misbehave.

Reply:
Ok!